src/state/hooks.ts

```diff
@@ -1,5 +1,7 @@
 import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
-import { AppDispatch, AppState } from 'state'
 
-export const useAppDispatch = () => useDispatch<AppDispatch>()
+import store from './index'
+import { AppState } from './types'
+
+export const useAppDispatch = () => useDispatch<typeof store.dispatch>()
 export const useAppSelector: TypedUseSelectorHook<AppState> = useSelector
```

Comment:
nit: this could be `TypedUseSelectorHook<ReturnType<store.getState>>` to avoid bringing in `./types`.